engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetCapability.java

```diff
@@ -17,46 +17,50 @@
 
 import org.terasology.rendering.dag.RenderPipelineTask;
 import org.terasology.rendering.dag.StateChange;
-import org.terasology.rendering.dag.tasks.DisableCapabilityTask;
 
 /**
  * TODO: Add javadocs
+ * Indented for capabilities that are enabled/disabled via glEnable and glDisable.
  */
-public class DisableCapability implements StateChange {
-    private int capability;
-    private DisableCapabilityTask task;
+public abstract class SetCapability implements StateChange {
+    private boolean enabled;
 
-    public DisableCapability(int capability) {
-        this.capability = capability;
+    public SetCapability(boolean enabled) {
+        this.enabled = enabled;
     }
 
-    public int getCapability() {
-        return capability;
+    @Override
+    public boolean isEqualTo(StateChange stateChange) {
+        if (stateChange instanceof SetCapability) {
+            return this.enabled == ((SetCapability) stateChange).isEnabled();
+        }
+        return false;
     }
 
-    @Override
-    public StateChange getDefaultInstance() {
-        return null;
+    public boolean isEnabled() {
+        return enabled;
     }
 
     @Override
     public RenderPipelineTask generateTask() {
-        if (task == null) {
-            task = new DisableCapabilityTask(capability);
+        if (isEnabled()) {
+            return getEnablingTask();
+        } else {
+            return getDisablingTask();
```

Comment:
In line with the comment above you can remove the getEnabling/DisablingTask calls and simply replace them with return enablingTask and return disablingTask. I'd also just check for if(enabled) and eliminate isEnabled() unless there is another reason to keep it.